xla/hlo/ir/hlo_instructions.cc

```diff
@@ -917,6 +917,7 @@ HloAllReduceInstructionBase::HloAllReduceInstructionBase(
                                constrain_layout, channel_id),
       use_global_device_ids_(use_global_device_ids) {
   AppendComputation(reduce_computation);
+  reduce_computation->SetCollectiveCallInstruction(this);
```

Comment:
I am assuming this covers both AR and RS. Can we add a test as well? in hlo_instruction_test.cc

Reply:
Yes, added tests for each op